lib/_http_client.js

```diff
@@ -711,11 +732,15 @@ ClientRequest.prototype.onSocket = function onSocket(socket) {
 };
 
 function onSocketNT(req, socket) {
-  if (req.aborted) {
+  if (req.destroyed) {
+    const { destroyCallback: cb } = req[kWritableState];
     // If we were aborted while waiting for a socket, skip the whole thing.
     if (!req.agent) {
-      socket.destroy();
+      socket.destroy(null, cb);
```

Comment:
Can we avoid destroy() call with undocumented cb param? E.g.:
```suggestion
      finished(socket, cb);
      socket.destroy();
```

Reply:
Unfortunate no. The callback must be invoked and in the correct order.